### DISCUSSION TITLE
"catch-all" namespace handler does not work

### DISCUSSION THREAD

**Discussion by Champgy** [miguelgrinberg/python-socketio#1315] [Q&A] [answered]
Hi,
I had a problem where I have multiple namespaces associated with the same handler, and the namespaces can be dynamic (s1, s2, s3, ..., sn). So, I tried using a "catch-all" namespace handler. 

**Problem**
I found that when I defined the connect event explicitly for a specific namespace, the "catch-all" namespace handler doesn't work.

Here is code.
```
from aiohttp import web
import socketio

sio = socketio.AsyncServer()
app = web.Application()
sio.attach(app)


@sio.on('connect')
async def connect(sid, data ):
    print("connect ", sid)

@sio.on('connect', namespace="/s1")
async def connect(namespace, sid, data ):
    print("connect ", sid)

@sio.on('connect', namespace="/s2")
async def connect(namespace, sid, data ):
    print("connect ", sid)

@sio.on('my_event', namespace="*")
async def my_event_any_namespace(namespace, sid, data):
    print(namespace, sid,  data)


if __name__ == '__main__':
    web.run_app(app, port=8000)

```

I observed that in the get_event_handler function, handlers[namespace] only contains the connect event, while my_event is located in handlers['*']. This ultimately leads to the handler being None. 
https://github.com/miguelgrinberg/python-socketio/blob/fff4640616e0516d7d6a72587d132257a8b780a2/src/socketio/base_server.py#L222-L228

Question
1. Can the following condition be changed to `if handler is None and '*' in self.handlers:`
https://github.com/miguelgrinberg/python-socketio/blob/fff4640616e0516d7d6a72587d132257a8b780a2/src/socketio/base_server.py#L229

**Comment by miguelgrinberg** [comment, marked as answer]
Yes, this change seems reasonable. Do you want to submit a PR?
